components/webext-storage/src/migration.rs

```diff
@@ -107,29 +108,42 @@ struct Parsed<'a> {
     data: serde_json::Value,
 }
 
-pub fn migrate(tx: &Transaction<'_>, filename: &Path) -> Result<usize> {
+pub fn migrate(tx: &Transaction<'_>, filename: &Path) -> Result<MigrationInfo> {
     // We do the grouping manually, collecting string values as we go.
     let mut last_ext_id = "".to_string();
     let mut curr_values: Vec<(String, serde_json::Value)> = Vec::new();
-    let mut num_extensions = 0;
-    for row in read_rows(filename) {
+    let mut curr_quota_bytes: usize = 0;
+    let (rows, mut mi) = read_rows(filename);
+    for row in rows {
         log::trace!("processing '{}' - '{}'", row.col_name, row.record);
         let parsed = match row.parse() {
             Some(p) => p,
-            None => continue,
+            None => {
+                mi.non_fatal_errors_seen += 1;
```

Comment:
How do you feel about splitting `non_fatal_errors` into `database_errors` and `parse_errors`? `open_failure` is worth calling out separately either way, but I wonder if there's room to distinguish between the database file itself being weird, and having a database that's structurally fine, but contains invalid data.

Reply:
I've killed `non_fatal_errors`, we just have `open_failure` and the counts now.